File: app/src/main/java/com/github/st1hy/countthemcalories/activities/settings/view/SettingsActivity.java
package com.github.st1hy.countthemcalories.activities.settings.view;

import android.os.Bundle;

import com.github.st1hy.countthemcalories.R;
import com.github.st1hy.countthemcalories.activities.settings.fragment.view.SettingsFragment;
import com.github.st1hy.countthemcalories.activities.settings.inject.SettingsActivityModule;
import com.github.st1hy.countthemcalories.core.baseview.BaseActivity;
import com.github.st1hy.countthemcalories.core.drawer.DrawerPresenter;

import javax.inject.Inject;

public class SettingsActivity extends BaseActivity {

    @Inject
    DrawerPresenter drawerPresenter;
    @Inject
    SettingsFragment settingsFragment; //injects component

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.settings_activity);
        getAppComponent().newSettingsActivityComponent(new SettingsActivityModule(this))
                .inject(this);
    }

    @Override
    protected void onStart() {
        super.onStart();
        drawerPresenter.onStart();
    }

    @Override
    protected void onStop() {
        super.onStop();
        drawerPresenter.onStop();
    }
}

File: ui/src/main/java/com/github/st1hy/countthemcalories/ui/activities/settings/view/SettingsActivity.kt
package com.github.st1hy.countthemcalories.ui.activities.settings.view

import android.os.Bundle
import android.support.v4.app.Fragment
import com.github.st1hy.countthemcalories.ui.R
import com.github.st1hy.countthemcalories.ui.core.baseview.BaseActivity
import com.github.st1hy.countthemcalories.ui.core.drawer.DrawerPresenter
import com.github.st1hy.countthemcalories.ui.inject.core.ActivityModule
import javax.inject.Inject

internal class SettingsActivity : BaseActivity() {

    @Inject lateinit var drawerPresenter: DrawerPresenter
    @Inject lateinit var fragments: Map<String, Fragment> //injects component

    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.settings_activity)
        getAppComponent().newSettingsActivityComponent(ActivityModule(this))
                .inject(this)
    }

    override fun onStart() {
        super.onStart()
        drawerPresenter!!.onStart()
    }

    override fun onStop() {
        super.onStop()
        drawerPresenter!!.onStop()
    }

    override fun onBackPressed() {
        if (drawerPresenter!!.onBackPressed()) super.onBackPressed()
    }
}